playback/src/audio_backend/rodio.rs

```diff
@@ -0,0 +1,96 @@
+use super::{Open, Sink};
+extern crate rodio;
+use std::{io, thread, time};
+use std::process::exit;
+
+pub struct RodioSink {
+    rodio_sink: rodio::Sink,
+}
+
+fn list_outputs() {
+    println!("Default Audio Device:\n  {:?}", rodio::default_output_device().map(|e| e.name()));
+
+    println!("Available Audio Devices:");
+    for device in rodio::output_devices() {
+        println!("- {}", device.name());
+        // Output formats
+        if let Ok(fmt) = device.default_output_format() {
+            println!("  Default format:\n    {:?}", fmt);
+        }
+        let mut output_formats = match device.supported_output_formats() {
+            Ok(f) => f.peekable(),
+            Err(e) => {
+                println!("Error: {:?}", e);
+                continue;
+            },
+        };
+        if output_formats.peek().is_some() {
+            println!("  All formats:");
+            for format in output_formats {
+                println!("    {:?}", format);
+            }
+        }
+    }
+}
+
+impl Open for RodioSink {
+    fn open(device: Option<String>) -> RodioSink {
+        info!("Using rodio sink");
+
+        let mut rodio_device = rodio::default_output_device().expect("no output device available");
+        if device.is_some() {
+            let device_name = device.unwrap();
+
+            if device_name == "?".to_string() {
+                list_outputs();
+                exit(0)
+            }
+            let mut found = false;
+            for d in rodio::output_devices() {
+                if d.name() == device_name {
+                    rodio_device = d;
+                    found = true;
+                    break;
+                }
+            }
+            if !found {
```

Comment:
why don't we just put this block in an else statement if device name not found and lose the found variable alltogether.

Reply:
Ignore this, missed the fact it was in a for loop.